### PR TITLE
Bugfix 'shift-count-overflow' compilation error.

### PR DESCRIPTION
**Pull request by rafacouto**
It arranges this compilation error:

```
In file included from /home/caligari/src/simba/simba/src/drivers/network/uart.h:35:0,
                 from /home/caligari/src/simba/simba/src/simba.h:164,
                 from /home/caligari/src/simba/simba/src/drivers/network/uart.c:31:
/home/caligari/src/simba/simba/src/drivers/network/uart.c: In function 'uart_init':
/home/caligari/src/simba/simba/src/drivers/ports/stm32f1/uart_port.h:42:40: error: left shift count >= width of type [-Werror=shift-count-overflow]
 #define STM32_UART_DATA_BITS_8  ((0x00 << STM32_USART_CR1_M) >> 8)
```

### CODE REVIEW

File: src/drivers/ports/stm32f1/uart_port.h [hunk 1]
```diff
@@ -31,7 +31,44 @@
...
 
-#define UART_PORT_FRAME_FORMAT_DEFAULT 0
+/*
+ * STM32F1xx product line supports 8/9 data bits, N/E/O parity, 1/2 stop bits
+ * NOTE format field codification:
+ *      bits 0-7 are 8-15 of CR1 (coded 8 bits to the right)
+ *      bits 8-15 are 8-15 of CR2
+ */
+#define STM32_UART_DATA_BITS_8  0x00
+#define STM32_UART_DATA_BITS_9  ((0x01 << STM32_USART_CR1_M) >> 8)
```
Comment (eerimoq) [inline src/drivers/ports/stm32f1/uart_port.h]: ``STM32_USART_CR1_M`` is defined as ``BIT(12)``, which expands to ``1 << 12``, or ``4096`` decimal. So the compilation error persists selecting a non-default frame format. Please fix the compilation error for all frame formats.

Comment (rafacouto) [inline src/drivers/ports/stm32f1/uart_port.h]: Oppps 👼

File: src/drivers/ports/stm32f1/uart_port.h [hunk 2]
```diff
@@ -31,7 +31,44 @@
...
+ *      bits 8-15 are 8-15 of CR2
+ */
+#define STM32_UART_DATA_BITS_8  0x00
+#define STM32_UART_DATA_BITS_9  ((0x01 << STM32_USART_CR1_M_POS) >> 8)
+#define STM32_UART_PARITY_N     0x00
+#define STM32_UART_PARITY_E     ((0x02 << STM32_USART_CR1_PS_POS) >> 8)
+#define STM32_UART_PARITY_O     ((0x03 << STM32_USART_CR1_PS_POS) >> 8)
+#define STM32_UART_STOP_BITS_1  0x00
+#define STM32_UART_STOP_BITS_2  (0x02 << STM32_USART_CR2_STOP_POS)
+#define UART_PORT_FRAME_FORMAT_8N1 (STM32_UART_DATA_BITS_8 \
```
Comment (eerimoq) [inline src/drivers/ports/stm32f1/uart_port.h]: I find it much easier to understand using the MCU defines like this:

```
#define STM32_UART_DATA_BITS_8  (0)
#define STM32_UART_DATA_BITS_9  (STM32_USART_CR1_M >> 8)
#define STM32_UART_PARITY_N     (0)
#define STM32_UART_PARITY_E     (STM32_USART_CR1_PCE >> 8)
#define STM32_UART_PARITY_O     ((STM32_USART_CR1_PCE | STM32_USART_CR1_PS) >> 8)
#define STM32_UART_STOP_BITS_1  STM32_USART_CR2_STOP(0)
#define STM32_UART_STOP_BITS_2  STM32_USART_CR2_STOP(2)
```

Comment (rafacouto) [inline src/drivers/ports/stm32f1/uart_port.h]: Well, CR2_STOP is a 2-bits field and it matches with single bit defines. Using _POS is the pattern I have followed in other MCUs defines. Anyway, the result is indifferent. Please, modify the code as you believe it is the right way 😉

Comment (eerimoq) [inline src/drivers/ports/stm32f1/uart_port.h]: Here is an example of setting a bit field:

https://github.com/eerimoq/simba/blob/master/src/mcus/stm32f100rb/stm32.h#L93-L95

**Pull request closed without merging**